frontends/p4/p4-parse.ypp

```diff
@@ -993,6 +993,18 @@ const IR::P4Program *parse_P4_16_file(const char *name, FILE *in) {
     return new IR::P4Program(declarations->srcInfo, declarations);
 }
 
+const IR::P4Program *parse_string(std::string &pgm) {
```

Comment:
`const std::string&` is probably preferable, since it doesn't look like you mutate `pgm` in here.